test/e2e/accounts/common.ts

```diff
@@ -196,17 +188,36 @@ async function switchToAccount2(driver: Driver) {
 
 export async function connectAccountToTestDapp(driver: Driver) {
   await switchToOrOpenDapp(driver);
-
   await driver.clickElement('#connectButton');
 
-  await driver.delay(regularDelayMs);
   await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
-  await driver.clickElementAndWaitForWindowToClose({
+
+  const edit = await driver.findClickableElements({
+    text: 'Edit',
+    tag: 'button',
+  });
+  await edit[1].click();
```

Comment:
since there are 2 elements with the exact same test id and text, we use findClickableElements and then click. Since findClickableElements already makes sure that the element is clickable, we won't run into any race condition where we try to click an element that is not clickable